src/achievements.hpp

```diff
@@ -0,0 +1,73 @@
+/*
+	Copyright (C) 2003 - 2022
+	by David White <dave@whitevine.net>
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "config.hpp"
+
+struct achievement
+{
+	std::string id_;
+	std::string name_;
+	std::string name_completed_;
+	std::string description_;
+	std::string description_completed_;
```

Comment:
Several of these look like they should be `tstring`s.

Reply:
Yes, all of these four should certainly be `t_string`.

The icon should not be; it can be localized in a different manner if really needed, though I don't think it should need localization at all.

Reply:
Updated.